app/code/Magento/Elasticsearch/SearchAdapter/Aggregation/Builder/Dynamic.php

```diff
@@ -77,11 +77,7 @@ private function prepareData($data)
     {
         $resultData = [];
         foreach ($data as $value) {
-            $from = is_numeric($value['from']) ? $value['from'] : '*';
-            $to = is_numeric($value['to']) ? $value['to'] : '*';
-            unset($value['from'], $value['to']);
-
-            $rangeName = "{$from}_{$to}";
+            $rangeName = "{$value['from']}_{$value['to']}";
             $resultData[$rangeName] = array_merge(['value' => $rangeName], $value);
```

Comment:
This line is causing static failure due to array_merge in loop. can it be refactored to avoid that?